crates/extensions/tedge_flows/src/js_script.rs

```diff
@@ -116,24 +119,51 @@ impl JsScript {
         Ok(())
     }
 
+    /// Initialize the next execution time for this script's interval
+    /// Should be called after the script is loaded and interval is set
+    pub fn init_next_execution(&mut self) {
+        if !self.no_js_on_interval_fun && !self.interval.is_zero() {
+            self.next_execution = Some(Instant::now() + self.interval);
+        }
+    }
+
+    /// Check if this script should execute its interval function now
+    /// Returns true and updates next_execution if it's time to execute
+    pub fn should_execute_interval(&mut self) -> bool {
+        if self.no_js_on_interval_fun || self.interval.is_zero() {
+            return false;
+        }
+
+        let now = Instant::now();
```

Comment:
I would pass `now` as an argument of the method. Having the time externalized makes things easier to test. Using `tokio::time` mocking helps for unit tests, but is useless for `tedge flows test`